Model/Cron.php

```diff
@@ -829,6 +829,12 @@ protected function _retrieveLast4DigitsFromReason($reason)
      */
     protected function _holdCancelOrder($ignoreHasInvoice)
     {
+        $canCancel = $this->adyenHelper->getAdyenAbstractConfigData('notifications_can_cancel');
+        if(!$canCancel){
+            $this->_adyenLogger->addAdyenNotificationCronjob('Order can not be canceled based on the plugin configuration');
+            return;
+        }
+
```

Comment:
for OFFER_CLOSED, before calling **_holdCancelOrder**, in case the order cannot be cancelled, we set the state to STATE_NEW (line 1003).
Can we add the same check there for _notifications_can_cancel_ ? If we don't want to cancel the order, we also don't want to set its state to new

Reply:
Good catch. Since this also applies for ORDER_CLOSED we decided to create a new helper class for Config values. Now we're checking the notifications_can_cancel on ORDER_CLOSED and OFFER_CLOSED before changing the state to NEW.